engine/src/mbliphonegfx.mm

```diff
@@ -200,7 +200,6 @@ void UnlockPixels(MCGIntegerRectangle p_area, MCGRaster& p_raster, bool p_update
 		if (p_update)
 			FlushBits(p_area, p_raster . pixels, p_raster . stride);
 		
-		free(p_raster . pixels);
```

Comment:
I see that I'm leaking at the moment here in the case of `! p_update`.. should now be:
```
if (p_update)
{
		FlushBits(p_area, p_raster . pixels, p_raster . stride);
}
else
{
		free(p_raster . pixels);
}
```

Reply:
@runrevmark perhaps `FlushBits` should set a boolean indicating if it has taken ownership of the bits? Then we can do:

```
bool t_taken = false;
if (p_update)
 		FlushBits(p_area, p_raster . pixels, p_raster . stride, t_taken);

if (!t_taken)
 		free(p_raster . pixels);
```

Reply:
@runrevmark I'm running with the above as it means I can only set x_taken if the raster is successfully converted to a CGImage. Will push up in a bit.